addon/i3dio/ui/object.py

```diff
@@ -752,144 +843,42 @@ def execute(self, context):
-            ('z_axis_drive',),
-            ('drive_position',),
-            ('projection_distance',),
-            ('projection_angle',),
-            ('drive_force_limit',),
-            ('drive_spring',),
-            ('drive_damping',),
-            ('breakable_joint',),
-            ('joint_break_force',),
-            ('joint_break_torque',)
-        ]
-
-        for prop in properties:
-            row = layout.row()
-            row.prop(obj.i3d_attributes, prop[0])
-            if obj.i3d_attributes.joint is False:
-                row.enabled = False
-                obj.i3d_attributes.property_unset(prop[0])
-
-
-@register
-class I3D_IO_PT_reference_file(Panel):
-    bl_space_type = 'PROPERTIES'
-    bl_region_type = 'WINDOW'
-    bl_label = 'Reference File'
-    bl_context = 'object'
-    bl_parent_id = 'I3D_IO_PT_object_attributes'
-
-    @classmethod
-    def poll(cls, context):
-        return context.object is not None and context.object.type == 'EMPTY'
-
-    def draw(self, context):
-        layout = self.layout
-        layout.use_property_split = True
-        layout.use_property_decorate = False
-        layout.prop(context.object, 'i3d_reference_path')
-
-
-@register
-class I3DMappingData(bpy.types.PropertyGroup):
-    is_mapped: BoolProperty(
-        name="Add to mapping",
-        description="If checked this object will be mapped to the i3d mapping of the xml file",
-        default=False
-    )
-
-    mapping_name: StringProperty(
-        name="Alternative Name",
-        description="If this is left empty the name of the object itself will be used",
-        default=''
-    )
-
-@register
-class I3D_IO_PT_mapping_attributes(Panel):
-    bl_space_type = 'PROPERTIES'
-    bl_region_type = 'WINDOW'
-    bl_label = "I3D Mapping"
-    bl_context = 'object'
-    bl_parent_id = 'I3D_IO_PT_object_attributes'
-
-    @classmethod
-    def poll(cls, context):
-        return context.object is not None
-
-    def draw(self, context):
-        layout = self.layout
-        layout.use_property_split = True
-        layout.use_property_decorate = False
-        obj = bpy.context.active_object
-
-        row = layout.row()
-        row.prop(obj.i3d_mapping, 'is_mapped')
-        row = layout.row()
-        row.prop(obj.i3d_mapping, 'mapping_name')
-
 
 def register():
     for cls in classes:
         bpy.utils.register_class(cls)
     bpy.types.Object.i3d_attributes = PointerProperty(type=I3DNodeObjectAttributes)
-    bpy.types.Object.i3d_merge_group_index = IntProperty(default = -1)
+    bpy.types.Object.i3d_merge_group_index = IntProperty(default=-1)
     bpy.types.Object.i3d_mapping = PointerProperty(type=I3DMappingData)
-    bpy.types.Object.i3d_reference_path = StringProperty(
-        name="Reference Path",
-        description="Put the path to the .i3d file you want to reference here",
-        default='',
-        subtype='FILE_PATH')
+    bpy.types.Object.i3d_reference = PointerProperty(type=I3DReferenceData)

```

Comment:
This just drops any references specified in the old format instead of converting the format. I would suggest doing a conversion similar to what is done for [old merge groups](https://github.com/NMC-TBone/I3D-Blender-Addon/blob/7e261661ca66ddb2f3f992744b9214dbd8a913a9/addon/i3dio/ui/object.py#L848-L849)

Reply:
Hmm, have enough people actually used that feature previously that it make sense to do a conversion for it?

Reply:
> Hmm, have enough people actually used that feature previously that it make sense to do a conversion for it?

I have no way of knowing that. But in general I don't like breaking peoples Blender files if there is a way to do an upgrade seamlessly.

Reply:
Okay I will add a handler for it then, just in case 🙈